### PR TITLE
[stm32] H7 DMA + DMA data fixes + fix DAC DMA

### PR DESCRIPTION
**Pull request by chris-durand**
This adds support for H7 DMA. The device data [has to be fixed first](https://github.com/modm-io/modm-devices/issues/74). H7 devices use the same DMA IP as F4/F7 devices but use the DMAMUX peripheral for request multiplexing.

The updates in modm-devices fix broken DMA request data for many G0, L4, H7 and WL devices.

- [x] Adapt DMA driver for H7
- [x] Fix requests mappings in modm-devices data
- [x] Handling for L4+ DMAMUX data with shared request for two peripherals (DCMI/PSSI)
- [x] Fix DAC DMA driver for F2, F4, F7, H7
- [x] Add DAC DMA example if missing and test
   - [x] G4
   - [x] H7
   - [x] L4
   - [x] F4
   - [x] F7
- [x] Test H7 DMA

### CODE REVIEW

**Comment by chris-durand** [comment]
DMA is working on H7 but I there is an issue with the DAC DMA driver on H7 and, most likely, also on F2, F4 and H7.

On STMG4 the DAC DMA driver only works if the 16 bit samples are written as a 32 bit transfer. When both memory and peripheral sizes are set to "Half-Word" (16 bit) nothing happens.
The stream-channel DMA IP will however force FIFO mode when peripheral and data sizes don't match. Thus, it writes one sample to each half of the data register instead of zero-extending the 16 bit sample to 32 bit as it happens on STM32G4. That results in only every second sample being output by the DAC and the second half of the output is garbled data. 

I will add DAC DMA examples for every STM32 series I have a devboard at home of, test what works and fix the driver.

**Comment by chris-durand** [comment]
DAC DMA is now also working on F4. F3 is not working yet, L4 and F7 are untested.

**Comment by chris-durand** [comment]
@salkinium If the CI succeeds, this is ready to be merged for the release. H7 DMA is working fine and I tested the DAC DMA driver on all platforms listed above. As I don't have the F3 board with me I could not test that one. The [device data fix](https://github.com/modm-io/modm-devices/pull/75) should go first.

File: examples/nucleo_f446ze/dac_dma/main.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	Timer4::setMode(Timer4::Mode::UpCounter);
+	Timer4::setPrescaler(1);
+	Timer4::setOverflow(90 - 1);
+	Timer4::applyAndReset();
+	Timer4::start();
+
+	// Enable trigger out for timer 4
+	TIM4->CR2 |= TIM_CR2_MMS_1;
+
+	//GpioOutputA4::setOutput(true);
```
Comment (chris-durand) [inline examples/nucleo_f446ze/dac_dma/main.cpp]: This line should not be here

**Review by salkinium — APPROVED** [review]
Very nice as always! I'll merge it tomorrow afternoon.

**Pull request merged**